crates/circuits/types/src/batch/public_inputs.rs

```diff
@@ -137,18 +128,15 @@ impl PublicInputs for VersionedBatchInfo {
     /// - state roots MUST be chained
     /// - batch hashes MUST be chained
     /// - L1 msg queue hashes MUST be chained
-    fn validate(&self, prev_pi: &Self) {
-        assert_eq!(self.1, prev_pi.1);
-        assert_eq!(self.0.chain_id, prev_pi.0.chain_id);
-        assert_eq!(self.0.parent_state_root, prev_pi.0.state_root);
-        assert_eq!(self.0.parent_batch_hash, prev_pi.0.batch_hash);
-        assert_eq!(self.0.prev_msg_queue_hash, prev_pi.0.post_msg_queue_hash);
-
-        if self.1 == ForkName::EuclidV1 {
-            assert_eq!(self.0.prev_msg_queue_hash, B256::ZERO);
-            assert_eq!(prev_pi.0.prev_msg_queue_hash, B256::ZERO);
-            assert_eq!(self.0.post_msg_queue_hash, B256::ZERO);
-            assert_eq!(prev_pi.0.post_msg_queue_hash, B256::ZERO);
+    fn validate(&self, prev_pi: &Self, fork_name: ForkName) {
+        assert_eq!(self.chain_id, prev_pi.chain_id);
+        assert_eq!(self.parent_state_root, prev_pi.state_root);
+        assert_eq!(self.parent_batch_hash, prev_pi.batch_hash);
+        if fork_name != ForkName::EuclidV1 {
```

Comment:
this is incorrect. The values are `0` for `EuclidV1` so:

```suggestion
        if fork_name == ForkName::EuclidV1 {
```